openstack/objectstorage/v1/objects/requests.go

```diff
@@ -574,6 +594,10 @@ func CreateTempURL(c *gophercloud.ServiceClient, containerName, objectName strin
 	if err != nil {
 		return "", err
 	}
+	urlToBeSigned, err := tempURL(c, containerName, objectName)
```

Comment:
We need two different URLs: one with non-escaped names (which will be signed), and one with escaped names (which will be used as the actual URL to send back to the caller). Documented here: https://docs.openstack.org/swift/latest/api/temporary_url_middleware.html#hmac-signature-for-temporary-urls